ivy_tests/test_ivy/test_frontends/test_jax/test_jax_lax_operators.py

```diff
@@ -1743,6 +1743,82 @@ def test_jax_lax_dot(
     )
 
 
+@st.composite
+def _get_dtype_inputs_for_batch_matmul(draw):
+    dim_size = draw(helpers.ints(min_value=1, max_value=4))
+    batch_size = draw(helpers.ints(min_value=1, max_value=3))
+    dtype = draw(helpers.get_dtypes("numeric", index=1, full=False))
+    if dim_size == 1:
+        lhs = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                shape=(
+                    dim_size,
+                    dim_size,
+                ),
+                min_value=2,
+                max_value=5,
+            )
+        )
+        rhs = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                shape=(
+                    dim_size,
+                    dim_size,
+                ),
+                min_value=2,
+                max_value=5,
+            )
+        )
+    else:
+        lhs = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                shape=(batch_size, dim_size, dim_size),
+                min_value=2,
+                max_value=5,
+            )
+        )
+        rhs = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                shape=(batch_size, dim_size, dim_size),
+                min_value=2,
+                max_value=5,
+            )
+        )
+    return dtype, lhs, rhs
+
```

Comment:
I think the way you are using `dim_size` makes it so `lhs` and `rhs` are always stacked square arrays. We should be testing for all possible shape cases.